torch/_dynamo/variables/functions.py

```diff
@@ -639,3 +639,88 @@ def as_python_constant(self):
                 *[arg.as_python_constant for arg in self.args],
                 **{k: v.as_python_constant() for k, v in self.keywords.items()},
             )
+
+
+class TritonKernelVariable(VariableTracker):
+    def __init__(self, kernel, grid, **kwargs):
+        super().__init__(**kwargs)
+        self.kernel = kernel
+        self.grid = grid
+
+    def call_function(
+        self, tx, args: "List[VariableTracker]", kwargs: "Dict[str, VariableTracker]"
+    ) -> "VariableTracker":
+        def call_kernel(grid, *args, **kwargs):
+            self.kernel.run(*args, grid=grid, **kwargs)
+
+        from .dicts import ConstDictVariable
+        from .lists import BaseListVariable
+
+        grid = self.grid
+
+        # If the grid is a function, then lets execute it and convert it to
+        # a list
+        if isinstance(grid, (NestedUserFunctionVariable, UserFunctionVariable)):
+            # Populate the special "meta" argument to call the grid function
+            # Triton functions cannot take variadic or default arguments,
+            # so iterating over the arguments and parameters in a linear
+            # fashion is enough.
+            d = {}
+            for idx, name in enumerate(self.kernel.arg_names):
+                # no duplicates
+                assert name not in d
+
+                if idx < len(args):
+                    d[name] = args[idx]
+                else:
+                    assert name in kwargs
+                    d[name] = kwargs[name]
```

Comment:
Some python trickery
```suggestion
            d = {**dict(zip(self.kernel.arg_names, args)), **kwargs}
```
(this will also help in the future when you do autotuning, since not all args will be provided)